src/server.cpp

```diff
@@ -938,25 +959,27 @@ static CTimingMeas JitterMeas ( 1000, "test2.dat" ); JitterMeas.Measure(); // TE
     }
 }
 
-void CServer::DecodeReceiveDataBlocks ( const int iStartChanCnt,
+void CServer::DecodeReceiveDataBlocks ( CServer*  sServer, 
```

Comment:
Will these pointers (here and the other calls) be thread-safe?  The loop above uses locking around its scope but it then kicks off multiple threads, passing each a reference to the same instance.  Have all the server methods used been checked for thread safety?